### DISCUSSION TITLE
RPU after demuxing shorter than videostream

### DISCUSSION THREAD

**Discussion by starcook** [quietvoid/dovi_tool#185] [General]
The problem: the RPU is from the same source but not frame-identical with the video stream, it runs faster(?).

In the first minutes everything is synchronous, at about half there is a delay of -8 frames and at the end even -14. The video itself runs at 23.976fps, the DV layer too, but comes to a different length.

The crazy thing: In the source it seems to work anyway. As soon as you demux and inject, it no longer works. The RPU is vanilla from the UHD but still 14 frames shorter than the video stream. How can that be?

I extracted the EL for fun and played with the fps, with 23.97475 it comes to the length of the video stream. Crazy.

Does anyone have any ideas? Is there any way to stretch the RPU? And why the hell is the source running correctly?!

IMG: https://abload.de/image.php?img=blaowiyn.png

**Comment by quietvoid** [comment]
Is your source a retail disc? If it is, what is the title?
Otherwise, it might just be an invalid file.

**Reply by starcook** [reply]
Yes, it is. Apocalypse Now (US-UHD, Redux).  On the disc the DV layer runs fine, but something seems to change when extracting.

**Comment by saindriches** [comment]
Did you get any warning during extraction? I think the RPU is not fully extracted.

**Reply by starcook** [reply]
No, nothing at all. Everything is running normally. I would understand if the RPU was just truncated, but it covers the entire video stream, just at a different speed.

**Comment by JonAbbott2** [comment]
I'm seeing a similar issue..same Bluray.  The RPU has a different number of frames to the video stream.

I've also seen this on two other Bluray:
Witness (1985) : https://www.blu-ray.com/movies/Witness-4K-Blu-ray/351329/
Mississippi Burning (1988) : https://www.blu-ray.com/movies/Mississippi-Burning-4K-Blu-ray/387492/

In both of these cases, the RPU is roughly half the number of frames.  Either lots of RPU frames were skipped when extracting, or DV is somehow relying on RPU frames spanning multiple video frames.

**Reply by quietvoid** [reply]
Both of those have known authoring issues, the two video streams are not in sync.
You didn't provide any info but MakeMKV is known to cause missing metadata for these titles.

**Reply by JonAbbott2** [reply]
Sorry, should have mentioned I used MakeMKV.  It's probably the same issue with Apocalypse Now Redux then.

I just tried imaging Apocalypse Now Final Cut US release, which has the same issue, the RPU is 12 frames longer than the film - I might order the STUDIOCANAL release of the Final Cut to see if its any different.

Something else I've just noticed with Apocalypse Now Redux.  If I extract the RPU from the muxed HEVC its 15 frames too long.  If I extract the EL, then extract the RPU from that, it's 15 frames short.

**Reply by JonAbbott2** [reply]
I can confirm MakeMKV is the cause of the issues with Apocalypse Now Final Cut.  If I decrypt the disk to an ISO, then demux the two video streams from the M2TS...and finally extract the RPU from the EL - its correct.

EDIT: Same process worked for Witness, so I'm fairly confident Apocalypse Now Remux and Missisippi Burning are the same.

Thanks for pointing me in the right direction...namely avoid MakeMKV if the BL/EL/RPU do not match
